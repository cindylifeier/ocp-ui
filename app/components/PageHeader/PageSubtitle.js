import styled from 'styled-components';

const PageSubtitle = styled.div`
  padding-left: 0.5vw;
  margin-top: 2vh;
  margin-left: 1vw;
  margin-right: 1vw;
  font-size: 1.2rem;
  font-weight: bold;
  color: #444;
  background-color: #f2f2f2;
`;

PageSubtitle.propTypes = {};

export default PageSubtitle;
