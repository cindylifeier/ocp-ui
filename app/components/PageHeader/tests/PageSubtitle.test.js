import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';

import PageSubtitle from '../PageSubtitle';

configure({ adapter: new Adapter() });

describe('<PageSubtitle />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<PageSubtitle>{children}</PageSubtitle>);

      // Arrange
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('structural tests', () => {
    it('should contain children', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<PageSubtitle>{children}</PageSubtitle>);

      // Arrange
      expect(renderedComponent.contains(children)).toEqual(true);
    });
  });

  describe('styling tests', () => {
    it('should have styles', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<PageSubtitle>{children}</PageSubtitle>);

      // Arrange
      expect(renderedComponent).toHaveStyleRule('padding-left', '0.5vw');
      expect(renderedComponent).toHaveStyleRule('margin-top', '2vh');
      expect(renderedComponent).toHaveStyleRule('margin-left', '1vw');
      expect(renderedComponent).toHaveStyleRule('margin-right', '1vw');
      expect(renderedComponent).toHaveStyleRule('font-size', '1.2rem');
      expect(renderedComponent).toHaveStyleRule('font-weight', 'bold');
      expect(renderedComponent).toHaveStyleRule('color', '#444');
      expect(renderedComponent).toHaveStyleRule('background-color', '#f2f2f2');
    });
  });
});
