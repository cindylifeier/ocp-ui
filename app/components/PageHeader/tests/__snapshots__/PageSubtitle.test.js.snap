// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageSubtitle /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageSubtitle__PageSubtitle>
    <div>
      test
    </div>
  </PageSubtitle__PageSubtitle>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        test
      </div>,
      "className": "PageSubtitle__PageSubtitle-bjOVNy dFgQid",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          test
        </div>,
        "className": "PageSubtitle__PageSubtitle-bjOVNy dFgQid",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
